extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -377,10 +385,27 @@ public void registerSwaggerUiHandler(SwaggerUiRecorder recorder,
             String oauthScopes = swaggerUiConfig.oauthScopes.get();
             options.put(Option.oauthScopes, oauthScopes);
         }
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, String> oauthAdditionalQueryStringParamMap = new HashMap<>();
         if (swaggerUiConfig.oauthAdditionalQueryStringParams.isPresent()) {
             String oauthAdditionalQueryStringParams = swaggerUiConfig.oauthAdditionalQueryStringParams.get();
-            options.put(Option.oauthAdditionalQueryStringParams, oauthAdditionalQueryStringParams);
+            Map<String, String> map = objectMapper.readValue(oauthAdditionalQueryStringParams, Map.class);
+            if (map == null || map.isEmpty()) {
+                LOG.warn(
+                        "Property 'quarkus.swagger-ui.oauth-additional-query-string-params' should be a map, example: quarkus.swagger-ui.oauth-additional-query-string-params='{\"foo\": \"bar\"}' ");
+            } else {
+                oauthAdditionalQueryStringParamMap.putAll(map);
```

Comment:
Why don't we make it a Map in the config instead of using JSON?

Reply:
Hi @gsmet , yes I thought about that, but that is a breaking change, so I decided to go with this. We can always deprecate this config and add one that has the map, so that older config still works. W.d.y.t?

Reply:
As long as it breaks cleanly (e.g. with an error message), I'd say we should fix the format and add a note to the migration guide.

But better do it in a separate PR anyway.